code/modules/wiremod/shell/brain_computer_interface.dm

```diff
@@ -34,96 +34,39 @@
 	else
 		return ..()
 
-/obj/item/circuit_component/bci_action
+/obj/item/circuit_component/equipment_action/bci
 	display_name = "BCI Action"
 	desc = "Represents an action the user can take when implanted with the brain-computer interface."
 	required_shells = list(/obj/item/organ/cyberimp/bci)
 
-	/// The icon of the button
-	var/datum/port/input/option/icon_options
-
-	/// The name to use for the button
-	var/datum/port/input/button_name
-
-	/// Called when the user presses the button
-	var/datum/port/output/signal
-
 	/// A reference to the action button itself
 	var/datum/action/innate/bci_action/bci_action
 
-/obj/item/circuit_component/bci_action/Initialize(mapload, default_icon)
-	. = ..()
-
-	if (!isnull(default_icon))
-		icon_options.set_input(default_icon)
-
-	button_name = add_input_port("Name", PORT_TYPE_STRING)
-
-	signal = add_output_port("Signal", PORT_TYPE_SIGNAL)
-
-/obj/item/circuit_component/bci_action/Destroy()
+/obj/item/circuit_component/equipment_action/bci/Destroy()
 	QDEL_NULL(bci_action)
 	return ..()
 
-/obj/item/circuit_component/bci_action/populate_options()
-	var/static/action_options = list(
-		"Blank",
-
-		"One",
-		"Two",
-		"Three",
-		"Four",
-		"Five",
-
-		"Blood",
-		"Bomb",
-		"Brain",
-		"Brain Damage",
-		"Cross",
-		"Electricity",
-		"Exclamation",
-		"Heart",
-		"Id",
-		"Info",
-		"Injection",
-		"Magnetism",
-		"Minus",
-		"Network",
-		"Plus",
-		"Power",
-		"Question",
-		"Radioactive",
-		"Reaction",
-		"Repair",
-		"Say",
-		"Scan",
-		"Shield",
-		"Skull",
-		"Sleep",
-		"Wireless",
-	)
-
-	icon_options = add_option_port("Icon", action_options)
-
-/obj/item/circuit_component/bci_action/register_shell(atom/movable/shell)
+/obj/item/circuit_component/equipment_action/bci/register_shell(atom/movable/shell)
+	. = ..()
 	var/obj/item/organ/cyberimp/bci/bci = shell
+	if(istype(bci))
+		bci_action = new(src)
+		update_action()
 
-	bci_action = new(src)
-	update_action()
-
-	bci.actions += list(bci_action)
+		bci.actions += list(bci_action)
 
-/obj/item/circuit_component/bci_action/unregister_shell(atom/movable/shell)
+/obj/item/circuit_component/equipment_action/bci/unregister_shell(atom/movable/shell)
 	var/obj/item/organ/cyberimp/bci/bci = shell
+	if(istype(bci))
+		bci.actions -= bci_action
+		QDEL_NULL(bci_action)
+	. = ..()
```

Comment:
```suggestion
	return ..()
```

Reply:
i'd commit these if I could but you seem to have that turned off so I have to wait for you to do this. It's a bit nitpicky but it's important to keep to these standards when you can even if they can get annoying to follow